py/dml/traits.py

```diff
@@ -444,7 +445,7 @@ def sort_method_implementations(implementations):
                             impl.name)
         rank_to_method[impl.rank] = impl
 
-    ranks = set(rank_to_method)
+    ranks = {r: None for r in rank_to_method}
```

Comment:
For other reading this:
As of python 3.7 dictionaries retain insertion order for keys _as part of the official spec_, so this is deterministic if rank_to_method order is deterministic